### PR TITLE
ADCM-3046 Add authentication audit for `login/` endpoint

### PR DESCRIPTION
**Pull request by a-alferov**

### CODE REVIEW

File: python/audit/middleware.py [hunk 1]
```diff
@@ -10,30 +10,22 @@
...
-            return self.get_response(request)
-
-        body = json.loads(request.body.decode('utf-8'))
-        username = body['username']
-        response = self.get_response(request)
-
-        if request.user.is_authenticated:
-            audit_user = request.user
+    @staticmethod
+    def _audit(user: User | AnonymousUser = None, username: str = None):
```
Comment (winmasta) [inline python/audit/middleware.py:23]: Maybe `Optional` needs to be used if `None` is default

File: python/audit/middleware.py [hunk 2]
```diff
@@ -10,30 +10,22 @@
...
+        """Authentication audit"""
+        if user.is_authenticated:
```
Comment (winmasta) [inline python/audit/middleware.py:25]: If `user` is `None` `AttributeError` will be raised

File: python/audit/middleware.py [hunk 3]
```diff
@@ -43,7 +35,18 @@ def __call__(self, request):
...
                 result = AuditSessionLoginResult.UserNotFound
+                user = None
+
+        AuditSession.objects.create(user=user, login_result=result, login_details=details)
+
+    def __call__(self, request):
+        response = self.get_response(request)
 
-        AuditSession.objects.create(user=audit_user, login_result=result, login_details=details)
+        if request.method == "POST" and request.path in [
```
Comment (winmasta) [inline python/audit/middleware.py:45]: nit: find in set is faster than in list

File: python/audit/middleware.py (hunk 1 above)
Comment (a-alferov) [reply python/audit/middleware.py]: fixed

Comment (a-alferov) [reply python/audit/middleware.py]: I use `Optional`, but maybe use `|`? For example `User | AnonymousUser | None`

File: python/audit/middleware.py (hunk 2 above)
Comment (a-alferov) [reply python/audit/middleware.py]: fixed

File: python/audit/middleware.py (hunk 3 above)
Comment (a-alferov) [reply python/audit/middleware.py]: fixed

**Review by winmasta — APPROVED** [review]

**Comment by a-alferov** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @a-alferov! Build has been launched!

**Review by winmasta — APPROVED** [review]

**Pull request merged**